internal/store/deployment.go

```diff
@@ -174,8 +174,8 @@ func deploymentMetricFamilies(allowAnnotationsList, allowLabelsList []string) []
 					for j, m := range conditionMetrics {
 						metric := m
 
-						metric.LabelKeys = []string{"condition", "status"}
-						metric.LabelValues = append([]string{string(c.Type)}, metric.LabelValues...)
+						metric.LabelKeys = []string{"reason", "condition", "status"}
+						metric.LabelValues = append([]string{c.Reason, string(c.Type)}, metric.LabelValues...)
```

Comment:
Since reason could theoretically be anything, could you provide a list of allowed reasons and only expose these? Similar to what we have for pods: https://github.com/kubernetes/kube-state-metrics/blob/main/internal/store/pod.go#L40

The list of valid deployment reasons can be found in: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/util/deployment_util.go#L67-L101

Reply:
Do you mean the output format will be like below?

As is
```
kube_deployment_status_condition{namespace="default",deployment="nginx",condition="Progressing",status="true"} 1
kube_deployment_status_condition{namespace="default",deployment="nginx",condition="Progressing",status="false"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",condition="Progressing",status="unknown"} 0
```

To be
```
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="NewReplicaSetAvailable",condition="Progressing",status="true"} 1
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="NewReplicaSetAvailable",condition="Progressing",status="false"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="NewReplicaSetAvailable",condition="Progressing",status="unknown"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="MinimumReplicasUnavailable",condition="Progressing",status="true"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="MinimumReplicasUnavailable",condition="Progressing",status="false"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="MinimumReplicasUnavailable",condition="Progressing",status="unknown"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="FoundNewReplicaSet",condition="Progressing",status="true"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="FoundNewReplicaSet",condition="Progressing",status="false"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="FoundNewReplicaSet",condition="Progressing",status="unknown"} 0
...
```

Reply:
qq: will one deployment in several conditions? Can we only keep 3 streams?

For example, only keep
```
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="NewReplicaSetAvailable",condition="Progressing",status="true"} 1
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="NewReplicaSetAvailable",condition="Progressing",status="false"} 0
kube_deployment_status_condition{namespace="default",deployment="nginx",reason="NewReplicaSetAvailable",condition="Progressing",status="unknown"} 0
```

Reply:
Checked cardinality keeps same. My comment has been resolved.